### DISCUSSION TITLE
great boilerplate. one question about the building process

### DISCUSSION THREAD

**Discussion by devhandler** [NekitCorp/chrome-extension-svelte-typescript-boilerplate#22] [General]
once we finish manifest.json, content.js, background.js, etc, does npm run dev or npm run build actually build/bundle the original scripts we write to /dist? or does it simply copy paste the original source code files to /dist without any bundling/building? i heard vite/rollup didn't support multi entry parallel building, so was wondering how the output prod files are linked to the source files. the vite.config.js file doesn't tell much.

**Comment by devhandler** [comment]
also, iiuc, content script doesn't allow import directly except dynamic import function. why import in your content script works?

```
import Overlay from "../components/Overlay.svelte";
import { storage } from "../storage";

// Content scripts
// https://developer.chrome.com/docs/extensions/mv3/content_scripts/

// Some global styles on the page
import "./styles.css";

// Some JS on the page
storage.get().then(console.log);

// Some svelte component on the page
new Overlay({ target: document.body });
```

**Reply by NekitCorp** [reply]
All the magic of building an extension using Vite is in the plugin `crx({ manifest })` from [@crxjs/vite-plugin](https://www.npmjs.com/package/@crxjs/vite-plugin) package. 

https://github.com/crxjs/chrome-extension-tools/tree/main/packages/vite-plugin

**Reply by devhandler** [reply]
makes sense. thank you!
